ngraph/frontend/paddlepaddle/src/model.cpp

```diff
@@ -228,6 +235,67 @@ namespace ngraph
 #endif
         } // namespace pdpd
 
+        std::vector<std::shared_ptr<OpPlacePDPD>>
+            InputModelPDPD::InputModelPDPDImpl::getOpPlaces() const
+        {
+            if (m_graph_changed)
+            {
+                return determine_cut_nodes();
+            }
+            return m_op_places;
+        }
+
+        std::vector<std::shared_ptr<OpPlacePDPD>>
+            InputModelPDPD::InputModelPDPDImpl::determine_cut_nodes() const
+        {
+            std::queue<OpPlacePDPD*> q;
+            std::unordered_set<OpPlacePDPD*> visited;
+            std::vector<std::shared_ptr<OpPlacePDPD>> new_op_places;
+            new_op_places.reserve(m_op_places.size());
+            // Marking nodes from outputs to inputs/constants
+            for (const auto& output : getOutputs())
+            {
+                if (!output->is_input())
```

Comment:
Is this check really needed? If yes, should there be an 'assert' instead?

Reply:
This check is needed to test if we have Parameter->Result graph, it would look strange, but don't think we should throw in that case.